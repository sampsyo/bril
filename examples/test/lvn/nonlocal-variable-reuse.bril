# CMD: bril2json < {filename} | python3 ../../lvn.py -p | bril2txt
@main {
  a: int = const 42;
.lbl:
  b: int = id a;
  # This should be re-named since `a` is defined outside of the local scope.
  a: int = const 5;
  print b;
}
