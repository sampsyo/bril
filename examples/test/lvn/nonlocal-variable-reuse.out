@main {
  a: int = const 42;
.lbl:
  b: int = id a;
  _a: int = const 5;
  print a;
}
